src/Config/RuntimeConfigLoader.cs

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Abstractions;
+using System.Net;
+using System.Reflection;
+using System.Text.Json;
+using Azure.DataApiBuilder.Config.Converters;
+using Azure.DataApiBuilder.Config.NamingPolicies;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Config;
+
+public class RuntimeConfigLoader
+{
+    private readonly IFileSystem _fileSystem;
+    private readonly string _baseConfigFileName;
+    private readonly string? _connectionString;
+    public const string CONFIGFILE_NAME = "dab-config";
+    public const string CONFIG_EXTENSION = ".json";
+
+    public const string ENVIRONMENT_PREFIX = "DAB_";
+    public const string RUNTIME_ENVIRONMENT_VAR_NAME = $"{ENVIRONMENT_PREFIX}ENVIRONMENT";
+    public const string RUNTIME_ENV_CONNECTION_STRING = $"{ENVIRONMENT_PREFIX}CONNSTRING";
+    public const string ASP_NET_CORE_ENVIRONMENT_VAR_NAME = "ASPNETCORE_ENVIRONMENT";
+
+    public static bool CheckPrecedenceForConfigInEngine = true;
+
+    public const string SCHEMA = "dab.draft.schema.json";
+
+    public string ConfigFileName => GetFileNameForEnvironment(Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT"), false);
+
+    public RuntimeConfigLoader(IFileSystem fileSystem, string baseConfigFileName = DEFAULT_CONFIG_FILE_NAME, string? connectionString = null)
+    {
+        _fileSystem = fileSystem;
+        _baseConfigFileName = baseConfigFileName;
+        _connectionString = connectionString;
+    }
+
+    /// <summary>
+    /// Load the runtime config from the specified path.
+    /// </summary>
+    /// <param name="path">The path to the dab-config.json file.</param>
+    /// <param name="config">The loaded <c>RuntimeConfig</c>, or null if none was loaded.</param>
+    /// <returns>True if the config was loaded, otherwise false.</returns>
+    public bool TryLoadConfig(string path, [NotNullWhen(true)] out RuntimeConfig? config)
+    {
+        if (_fileSystem.File.Exists(path))
+        {
+            string json = _fileSystem.File.ReadAllText(path);
```

Comment:
Wouldn't it make more sense to handle that higher up? Getting `false` back from this is an indicator that we failed to parse the config and maybe recoverable/handled by the callsite, but an `IOExpection` isn't something that can be recovered within the loader and higher up the call site we should handle an unhandled exception of the loader?

If we can't read the file due to permissions is there anything we can handle ?

Reply:
Perhaps not. Depends on how the exception is ultimately displayed to the user and whether it is swallowed by some other exception handlers unintentionally. This can be an added task to cleaning up some error logging / exception raising.